zerver/openapi/zulip.yaml

```diff
@@ -21041,6 +21041,23 @@ components:
         description: |
           `{config_key}`: Description/value of the configuration data key.
         type: string
+    IncomingBotConfiguration:
```

Comment:
I think this should be still called `Botconfiguration`, since existing non-incoming webhook bots can still use this framework? (embedded bots is the main category). Or is these something more subtle here, where you're just changing one user?

Also, you need an API feature level bump and API changelog entry and `**Changes**:` entry here documenting the API change from tuples to objects for this part of the API. Use `git log api_docs/` to find examples of how to do this, and refer to the new application feature tutorial for details.

Reply:
I want to update the bot configuration system to include fields like name, description, and validator. My goal is to introduce this without disrupting the existing BotConfiguration, which uses simpler key-value pairs.

Would it be appropriate to name this new structure WebhookConfigOption, or is there something else I should consider approach-wise?